### DISCUSSION TITLE
getServerSideProps

### DISCUSSION THREAD

**Discussion by nilaypatel251** [vercel/next.js#24927] [Help]
I have a next js frontend and a Graphql server on the backend, it works great with SSG but when I am using getServerSideProps it is making a request ( hitting the server ) only first time we visit any page and not on every refresh. Is this normal in SSR? ( Everything is in context of Production build)

**Comment by nilaypatel251** [comment]
export async function getServerSideProps() {


  const { data } = await client.query({
    query: gql`
      query Movies {
        movies{
          id
          name
          image
        }
      }
    `,
  });
  

  
  return {
    props: {
      parks: data.movies,
    },
 };
}



This is the ServerSideProp function

**Comment by fiskolini** [comment]
Did you find the answer? I am having the same issue

**Reply by nilaypatel251** [reply]
Yup, apparently cache was in play. I was using Apollo Client for making a request to the Endpoint. If you are using Apollo client check this link to docs. We can customize the fetch policy according to our requirements.

https://www.apollographql.com/docs/react/data/queries/#setting-a-fetch-policy

**Reply by fiskolini** [reply]
Thanks, it worked!
